src/main/java/org/quiltmc/loader/impl/transformer/RuntimeModRemapper.java

```diff
@@ -95,16 +92,35 @@ public static void remap(TransformCache cache) {
 				infoMap.put(mod, info);
 				InputTag tag = remapper.createInputTag();
 				info.tag = tag;
-				info.inputPath = mod.resourceRoot().toAbsolutePath();
 
-				if (requiresMixinRemap(info.inputPath)) {
+				if (requiresMixinRemap(mod.resourceRoot())) {
 					remapMixins.add(tag);
 				}
 
-				remapper.readInputsAsync(tag, info.inputPath);
+				Path in = fs.getPath(mod.id());
+				// HACK: Tiny Remapper eagerly opens ZIP files contained in mods (i.e. the JAR files they've attempted to JiJ)
+				// This causes LOTS of problems involving duplicate classes (and potentially the wrong version of the class being selected!!!),
+				// so we ONLY expose the .class files to Tiny Remapper
+				Files.walk(mod.resourceRoot()).filter(p -> p.getFileName().toString().endsWith(".class")).forEach(p -> {
+					try {
+						Files.createDirectories(in.resolve(p.getParent().toAbsolutePath().toString()));
+						fs.mount(p, in.resolve(p.toAbsolutePath().toString()));
```

Comment:
Adding a `MountOption.READ_ONLY` will avoid a partial FS copy duing `fs.switchToReadOnly` (`fs.mount(p, in.resolve(p.toAbsolutePath().toString()), MountOption.READ_ONLY);`)